stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -890,3 +890,38 @@ end
 *(A::Adjoint{<:Any,<:RealHermSymComplexHerm}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Adjoint{<:Any,<:AbstractTriangular}) = A.parent * B
 *(A::Transpose{<:Any,<:RealHermSymComplexSym}, B::Transpose{<:Any,<:AbstractTriangular}) = A.parent * B
+
+"""
+    symmetrize!(X::AbstractMatrix{<:AbstractFloat})
+
+Make the matrix `X` symmetric in-place using the formula `(X + X') / 2`.
+
+See also: [`symmetrize`](@ref)
+
+!!! compat "Julia 1.3"
+    This function requires Julia 1.3 or later.
+"""
+function symmetrize!(X::AbstractMatrix{<:AbstractFloat})
+    require_one_based_indexing(X)
+    n = checksquare(X)
+    @inbounds for j = 2:n, i = 1:j-1
```

Comment:
As mentioned in the comment, I think we should make this would for non-real matrices. In that case, you should include the diagonal do make this correct in the non-real case. Maybe move the diagonal outside the `i` loop and just use `real`.

Reply:
Is `real` the right thing? In `(Z + Z') / 2`, the diagonal is just the diagonal of `Z` but conjugated.

Reply:
Exactly ;)

Reply:
:man_facepalming: